### PR TITLE
domain: check schema change when lease=0

### PR DESCRIPTION
**Pull request by tiancaiamao**
When lease = 0, schemaValidator will always return true, which means commit won't check schema.
This introduce a bug that insert into table will success even the column is deleted by another transaction.
Localstore is affected by this bug, because its lease is 0.

Depends on another PR https://github.com/pingcap/tidb/pull/3270 to pass CI.

### CODE REVIEW

**Comment by zimulala** [comment]
Please fix CI.

File: session_test.go [hunk 1]
```diff
@@ -2730,3 +2730,20 @@ func (s *testSessionSuite) TestRetryResetStmtCtx(c *C) {
...
+
+	s2 := newSession(c, s.store, dbName)
+	mustExecSQL(c, s2, "begin")
+	mustExecSQL(c, s2, "insert into t values (1, 1)")
+
+	mustExecSQL(c, s1, "alter table t drop column b")
+
+	// When s2 commit, it will find schema already changed.
+	mustExecSQL(c, s2, "insert into t values (4, 4)")
+	mustExecFailed(c, s2, "commit")
```
Comment (zimulala) [inline session_test.go]: It's better to check the error message.

**Comment by zimulala** [comment]
Rest LGTM

**Comment by shenli** [comment]
LGTM

**Comment by shenli** [comment]
@tiancaiamao Please fix the CI.

**Pull request merged**